feg/gateway/services/n7_n40_proxy/n7/n7.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package n7
+
+import (
+	"context"
+	"net/http"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+
+	n7_client "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+)
+
+// createN7Client creates a N7 oapi client and sets the OAuth2 client credentiatls for authorizing requests
+func CreateN7Client(cfg *PCFConfig) (*n7_client.ClientWithResponses, error) {
+	httpClient := &http.Client{}
+	tokenConfig := clientcredentials.Config{
+		ClientID:     cfg.ClientId,
+		ClientSecret: cfg.ClientSecret,
+		TokenURL:     cfg.TokenUrl,
+	}
+	tokenCtxt := context.WithValue(context.Background(), oauth2.HTTPClient, httpClient)
+	doer := tokenConfig.Client(tokenCtxt)
+
+	serverStr, err := GetServerStringFromUrl(cfg.ApiRoot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create new N7 client object and assosiate it with oAuth2 HTTP client
+	client, err := n7_client.NewClientWithResponses(
+		serverStr,
+		n7_client.WithHTTPClient(doer),
+	)
```

Comment:
Create the variables closer where you are going to use them

Also try to reduce the usage of variables that are only used once. Unles it makes the code way more understandeabe

```
	tokenCtxt := context.WithValue(context.Background(), oauth2.HTTPClient, httpClient)
	client, err := n7_client.NewClientWithResponses(
		serverStr,
		n7_client.WithHTTPClient(tokenConfig.Client(tokenCtxt)),
	)

```